### DISCUSSION TITLE
document how to build / test translations

### DISCUSSION THREAD

**Discussion by axd1967** [Stellarium/stellarium#1205] [Q&A] [answered]
I try to improve the Dutch translations, but the branch containing the modified translations is not taken into account when rebuilding the code.

Wiki says to join "Transifex". All I want is to update the docstrings, such as in https://github.com/axd1967/stellarium/tree/contrib/translate-nl

Please explain how to include the modified files (`make translations`?) in the build? I currently do this:
```
cmake clean
cmake -DCMAKE_INSTALL_PREFIX=$PWD/ -DENABLE_GPS=0 -DENABLE_LIBGPS=0 -DCMAKE_BUILD_TYPE="RelWithDebInfo" $PROJECT
make clean translations
make -j6 install

```

thank you

**Comment by alex-w** [comment, marked as answer]
You need to follow Transifex docs for it, so, you need to use `tx` util or obtain `*.po` files through web-interface of Transifex and put `*.po` files into respectively directories.

For example with `tx` util:
```
cmake clean
tx pull -l nl
cmake -DCMAKE_INSTALL_PREFIX=$PWD/ -DENABLE_GPS=0 -DENABLE_LIBGPS=0 -DCMAKE_BUILD_TYPE="RelWithDebInfo" $PROJECT
make -j6 && make install
```

Translations are automatic updating on `make install` command.

P.S. Hmm.. why you do not use Transifex?

**Comment by axd1967** [comment]
omg... :nauseated_face:   

( I don't like all these web dependencies.)

**Comment by alex-w** [comment]
What about POEdit or Lokalize? You may upload your `*.po` files on Transifex...

**Comment by axd1967** [comment]
My mistake - I was building the wrong branch.

**Comment by axd1967** [comment]
(as I was fearing... this transifex website is slow...)

**Comment by axd1967** [comment]
when I want to translate "group of stars" - my input does not reach the server.

**Comment by axd1967** [comment]
@alex-w is it normal that Transifex is so slow?

**Comment by axd1967** [comment]
I try to translate "Ashes" to "Asse" but Transifex seems to stop working.

**Comment by axd1967** [comment]
god this is horrible!

**Comment by axd1967** [comment]
ok. I did a set of changes , how does this work?

**Comment by gzotti** [comment]
Alexander fetches translations on a regular basis and feeds them into our GIT.

**Comment by axd1967** [comment]
I managed to use the transifex site, as I hit a (predictable) merge conflict due to the fact that the po files seem to be generated, so I dropped the contribution via git for now.

however, there is one important change : there is an error, a switch between aquarius (should be "waterman") and aquila (should be "arend") in nl.po, I corrected this in transifex. (https://github.com/Stellarium/stellarium/blob/master/po/stellarium-skycultures/nl.po)
